### ISSUE TITLE
v1.38.0 breaks building on NetBSD

### ISSUE THREAD

**Issue by moubctez** [grpc/grpc#26357]
### What version of gRPC and what language are you using?
1.38.0

### What operating system (Linux, Windows,...) and version?
NetBSD

### What runtime / compiler are you using (e.g. python version or version of gcc)
GCC or Clang

gRPC 1.38.0 does not build on NetBSD. It looks like someone has forgotten to mention NetBSD in `include/grpc/event_engine/port.h`, which ends the build with `UNKNOWN PLATFORM`. This patch fixes the problem:

```
--- include/grpc/event_engine/port.h.orig	2021-05-25 05:21:42.595878017 +0000
+++ include/grpc/event_engine/port.h
@@ -22,7 +22,7 @@
 #elif defined(GPR_ANDROID) || defined(GPR_LINUX) || defined(GPR_APPLE) ||   \
     defined(GPR_FREEBSD) || defined(GPR_OPENBSD) || defined(GPR_SOLARIS) || \
     defined(GPR_AIX) || defined(GPR_NACL) || defined(GPR_FUCHSIA) ||        \
-    defined(GRPC_POSIX_SOCKET)
+    defined(GRPC_POSIX_SOCKET) || defined(GPR_NETBSD)
 #define GRPC_EVENT_ENGINE_POSIX
 #include <netdb.h>
 #include <netinet/in.h>
```

**Comment by jtattermusch** [comment]
@drfloob  since you recently added event_engine/port.h: https://github.com/grpc/grpc/pull/25886/files#diff-3a3574ff0d411c1f3085225b1416d9ea19007d79abce4d0f4475b212b3d0cb7e

**Comment by drfloob** [comment]
@moubctez Apologies, BSD is a bit of blind spot for our automated build/test infrastructure, we're looking into that. I'll do the fix and backport it to 1.38.0.

**Comment by drfloob** [comment]
Actually, we don't have explicit support for NetBSD. The NetBSD package maintainers keep a set of patches against gRPC that define `GPR_NETBSD` in port_platform, it's not defined in this repository at all. See https://github.com/NetBSD/pkgsrc/blob/trunk/net/grpc/patches/patch-src_core_lib_iomgr_port.h. It looks like they (maybe you? Someone named Adam) [already made this fix](https://github.com/NetBSD/pkgsrc/commit/0480364f8f55562be816566c67c4b15dffb6907b#diff-c3034c88f4dec936ff6bb5e2e650e0847760428833104926969e9d76db39427c) there.

gRPC-core added support for OpenBSD [in 2017](https://github.com/grpc/grpc/pull/12923), FreeBSD support [in 2015](https://github.com/grpc/grpc/pull/1047). It seems we should be able to add support for NetBSD as well, but again without the test harnesses set up in our CI, it probably wouldn't be wise, we'll continue to miss problems like this.

I'll see about fixing this for the other BSDs if needed.
